### DISCUSSION TITLE
Full self-hosting

### DISCUSSION THREAD

**Discussion by tmlmt** [steven-tey/novel#89] [Q&A] [answered]
Hey Steven, thanks for the great work !

Is it possible to deploy to one's own server i.e. without Vercel? Getting an OpenAI API Key is fine, but looking at the [README](https://github.com/steven-tey/novel#setting-up-locally), it seems Novel expects a Vercel Blob read/write token. Alternatively, what would it take to fully self-host Novel?

**Comment by z4nr34l** [comment, marked as answer]
It's just next.js project, you can deploy it however you want.
Here are the docs from Next.js that would help you: https://nextjs.org/docs/app/building-your-application/deploying

**Reply by tmlmt** [reply]
Cool, thank you. Might be worth adding this in the README.me for the sake of clarity, and also clarify whether the different environment variables are optional or not. They all seem to be optional as in:
- If `OPENAI_API_KEY` is not set, then you just won't be able to autocomplete with AI (PR #88 is a proposal to clarify a bit the error messages)
- The `VERCEL*` variables are also optional, as the `.env.example` suggests.
